GPUCollection/LLVMBitCodeVisitor.cs

```diff
@@ -187,12 +220,16 @@ protected override Expression VisitConstant(ConstantExpression c)
             {
                 switch (System.Type.GetTypeCode(c.Value.GetType()))
                 {
-                    case TypeCode.Boolean:
-                        // TODO: something meaningful
+                    case TypeCode.Int32:
+                        mainFunctionRefStack.Push(LLVM.ConstInt(LLVM.Int32Type(), Convert.ToUInt64((object)c.Value), new LLVMBool(0)));
                         break;
-                    case TypeCode.String:
-                        // TODO: something meaningful
-                        // sb.Append(c.Value);
+                    case TypeCode.Single:
+                        mainFunctionRefStack.Push(LLVM.ConstUIToFP(LLVM.ConstInt(LLVM.Int32Type(), Convert.ToUInt64((object)c.Value), new LLVMBool(0)), LLVM.FloatType()));
```

Comment:
You can use LLVM.ConstReal for constant floats and doubles

Reply:
Thanks for the tip, I wasn't sure how to do that properly